kernel/src/hil/adc.rs

```diff
@@ -2,37 +2,48 @@ use returncode::ReturnCode;
 
 /// Trait for handling callbacks from ADC module.
 pub trait Client {
-    /// Called when a sample is ready.
+    /// Called when a sample is ready. Used for single sampling
     fn sample_done(&self, sample: u16);
+
+    /// Called when the buffer is full. Used for continuous sampling.
+    /// The length provided will always be less than or equal to the length of the buffer.
+    /// Expects an additional call to either continue or stop sampling
+    fn buffer_ready(&self, buf: &'static mut [u16], length: usize);
```

Comment:
I think you may have somehow been looking at outdated code. If you check the current version of the adc hil (https://github.com/helena-project/tock/blob/22bd66937dbfdf16ffb91fd9ac42379778cd77d6/kernel/src/hil/adc.rs) you'll see that I have a `Client` trait with a `sample_ready` callback and a `HighSpeedClient` trait with `samples_ready` and a `sampling_complete`  callbacks.

Having these separate clients is necessary because while normal sampling (single or continuous) returns a single sample value at a time, high-speed sampling returns buffers of samples at a time. `sampling_complete` is used to return buffers to the higher-level capsule if an error occurs or after stop is called.

I'm confused about your concerns here, so if they are applicable to the current HIL, could you clarify what the issue is?